### PR TITLE
feature/bug: Add JSON schema + fix default pool bug

### PR DESCRIPTION
**Pull request by nickdnk**
Fix bug with missing defaults for pool
Only enable SSL if both cert and key is provided, not just one of them Fix wrong doc about unlimited MaxRequestSize (0 gets set to 1000 in default)

# Reason for This PR

Adding JSON schema.
Fix default pool init bug.

## Description of Changes

Adding a JSON schema in order to reference the pool from other repos.
Fixes missing initialization of defaults for the pool. If you provide only one property (say only `supervisor`), the rest does not get initialized to the default values.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the MIT license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -76,16 +74,9 @@ func (c *Config) EnableFCGI() bool {
...
-		// default pool
-		c.Pool = &pool.Config{
-			Debug:           false,
-			NumWorkers:      uint64(runtime.NumCPU()), //nolint:gosec
-			MaxJobs:         0,
-			AllocateTimeout: time.Second * 60,
-			DestroyTimeout:  time.Second * 60,
-			Supervisor:      nil,
-		}
+		c.Pool = &pool.Config{}
```
Comment (nickdnk) [inline config/config.go:77]: Unsure about the syntax here, so please review.

File: config/config.go [hunk 2]
```diff
@@ -62,7 +60,7 @@ func (c *Config) EnableTLS() bool {
 	if c.SSLConfig.Acme != nil {
 		return true
 	}
-	return c.SSLConfig.Key != "" || c.SSLConfig.Cert != ""
+	return c.SSLConfig.Key != "" && c.SSLConfig.Cert != ""
```
Comment (nickdnk) [inline config/config.go:63]: This is *technically* a breaking change, but I don't see how anyone can use SSL with only cert or only key. Both are required for it to work.

File: config/config.go [hunk 3]
```diff
@@ -27,7 +25,7 @@ type Config struct {
 	Pool *pool.Config `mapstructure:"pool"`
 	// InternalErrorCode used to override default 500 (InternalServerError) http code
 	InternalErrorCode uint64 `mapstructure:"internal_error_code"`
-	// MaxRequestSize specified max size for payload body in megabytes, set 0 to unlimited.
+	// MaxRequestSize specified max size for payload body in megabytes. 0 = 1GB.
```
Comment (nickdnk) [inline config/config.go:28]: There is no way to set unlimited body size with 0, as 0 gets converted to 1GB in init defaults.

File: config/config.go (hunk 2 above)
Comment (rustatian) [reply config/config.go:63]: Yep, correct, this is a bug

**Review by rustatian — APPROVED** [review]
LGTM, thanks @nickdnk 👍

**Pull request merged**